### PR TITLE
[receiver/k8scluster] Use newer batch and autoscaling APIs

### LINKED ISSUE #7386: [k8scluster receiver] Don't use deprecated batch/v1beta1 and autoscaling/v2beta1 Kubernetes API
Need to upgrade the k8scluster receiver code to use latest Kubernetes API to avoid warnings:

```
W0125 19:00:33.296477       1 warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
W0125 19:00:33.296975       1 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2beta2 HorizontalPodAutoscaler
W0125 19:00:33.313139       1 warnings.go:70] batch/v1beta1 CronJob is deprecated in v1.21+, unavailable in v1.25+; use batch/v1 CronJob
W0125 19:00:33.313282       1 warnings.go:70] autoscaling/v2beta1 HorizontalPodAutoscaler is deprecated in v1.22+, unavailable in v1.25+; use autoscaling/v2beta2 HorizontalPodAutoscaler
```

### PR DESCRIPTION
**Pull request by shree007**
**Description:** 
Since Kubernetes API  batch/v1beta1 and autoscaling/v2beta1 for Cronjob and Autoscaling has been deprecated  And its being used in receiver/k8scluster So It need to update with batch/v1 and autoscaling/v2beta2 respectively.

**Link to tracking Issue:** Closes #7386

### CODE REVIEW

**Comment by shree007** [comment]
@dmitryax  Please review 
cc: @jpkrohling

**Review by dashpole — APPROVED** [review]

**Comment by shree007** [comment]
since in collector.go and hpa.go imported library is not being used that's why some part of code build is being failed.
@jpkrohling  Should I remove unused imported library ?

**Comment by jpkrohling** [comment]
>  @jpkrohling Should I remove unused imported library ?

Yes

**Comment by shree007** [comment]
> > @jpkrohling Should I remove unused imported library ?
> 
> Yes

I think problem resides here 
It has been imported
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sclusterreceiver/internal/collection/collector.go#L26

But using as here without  giving reference https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sclusterreceiver/internal/collection/collector.go#L151

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/k8sclusterreceiver/internal/collection/collector.go#L188

So I just need to put 

`v2beta2 "k8s.io/api/autoscaling/v2beta2"` and replace v2beta2  with "v2beta1"

**Comment by shree007** [comment]
@dmitryax Please review.

**Review by jpkrohling — APPROVED** [review]

File: receiver/k8sclusterreceiver/internal/collection/collector.go [hunk 1]
```diff
@@ -23,9 +23,8 @@ import (
 	"go.opentelemetry.io/collector/model/pdata"
 	"go.uber.org/zap"
 	appsv1 "k8s.io/api/apps/v1"
-	"k8s.io/api/autoscaling/v2beta1"
+	v2beta2 "k8s.io/api/autoscaling/v2beta2"
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/collection/collector.go:26]: `v2beta2` is redundant alias, this is what k8s.io/api/autoscaling/v2beta2 gives anyway. If we want to provide an alias and avoid conflicts, I'd rename it to `autoscalingv2beta2`

File: receiver/k8sclusterreceiver/internal/collection/hpa_test.go [hunk 2]
```diff
@@ -19,7 +19,7 @@ import (
 
 	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
 	"github.com/stretchr/testify/require"
-	"k8s.io/api/autoscaling/v2beta1"
+	v2beta2 "k8s.io/api/autoscaling/v2beta2"
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/internal/collection/hpa_test.go:22]: Same here, redundant alias

File: receiver/k8sclusterreceiver/watcher.go [hunk 3]
```diff
@@ -28,9 +28,8 @@ import (
 	"go.uber.org/atomic"
 	"go.uber.org/zap"
 	appsv1 "k8s.io/api/apps/v1"
-	"k8s.io/api/autoscaling/v2beta1"
+	v2beta2 "k8s.io/api/autoscaling/v2beta2"
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/watcher.go:31]: And here

**Comment by dmitryax** [comment]
Please look at the failing tests, it maybe caused by changed behavior in the new versions

**Comment by shree007** [comment]
I will have a look into it

File: receiver/k8sclusterreceiver/internal/collection/collector.go (hunk 1 above)
Comment (shree007) [reply receiver/k8sclusterreceiver/internal/collection/collector.go:26]: sure. I will take care.

**Comment by shree007** [comment]
@dmitryax  I checked the test case there was some issue with `internal/aws/k8s` , I found that someone has already fixed the issue so I just merge main branch to my branch and I have updated the alias as you have suggested in comment. Please review
Thanks

cc: @jpkrohling

File: receiver/k8sclusterreceiver/watcher.go [hunk 4]
```diff
@@ -98,8 +97,8 @@ func (rw *resourceWatcher) prepareSharedInformerFactory() {
 	rw.setupInformers(&appsv1.ReplicaSet{}, factory.Apps().V1().ReplicaSets().Informer())
 	rw.setupInformers(&appsv1.StatefulSet{}, factory.Apps().V1().StatefulSets().Informer())
 	rw.setupInformers(&batchv1.Job{}, factory.Batch().V1().Jobs().Informer())
-	rw.setupInformers(&batchv1beta1.CronJob{}, factory.Batch().V1beta1().CronJobs().Informer())
-	rw.setupInformers(&v2beta1.HorizontalPodAutoscaler{},
+	rw.setupInformers(&batchv1.CronJob{}, factory.Batch().V1beta1().CronJobs().Informer())
+	rw.setupInformers(&autoscalingv2beta2.HorizontalPodAutoscaler{},
 		factory.Autoscaling().V2beta1().HorizontalPodAutoscalers().Informer(),
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/watcher.go:102]: I believe that informer passed as the second argument supposed to match the k8s object type

**Comment by shree007** [comment]
Ack.

File: receiver/k8sclusterreceiver/watcher.go (hunk 4 above)
Comment (shree007) [reply receiver/k8sclusterreceiver/watcher.go:102]: ack

File: receiver/k8sclusterreceiver/watcher.go [hunk 5]
```diff
@@ -98,9 +97,9 @@ func (rw *resourceWatcher) prepareSharedInformerFactory() {
 	rw.setupInformers(&appsv1.ReplicaSet{}, factory.Apps().V1().ReplicaSets().Informer())
 	rw.setupInformers(&appsv1.StatefulSet{}, factory.Apps().V1().StatefulSets().Informer())
 	rw.setupInformers(&batchv1.Job{}, factory.Batch().V1().Jobs().Informer())
-	rw.setupInformers(&batchv1beta1.CronJob{}, factory.Batch().V1beta1().CronJobs().Informer())
-	rw.setupInformers(&v2beta1.HorizontalPodAutoscaler{},
-		factory.Autoscaling().V2beta1().HorizontalPodAutoscalers().Informer(),
+	rw.setupInformers(&batchv1.CronJob{}, factory.Batch().V1beta1().CronJobs().Informer())
```
Comment (dmitryax) [inline receiver/k8sclusterreceiver/watcher.go:100]: Please update informer here as well

Comment (shree007) [reply receiver/k8sclusterreceiver/watcher.go]: ack @dmitryax

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**